controllers/socket/lobbyHandlers.go

```diff
@@ -217,8 +223,30 @@ func lobbyJoinHandler(so socketio.Socket) func(string) string {
 			chelpers.AfterLobbyJoin(so, lob, player)
 
 			if lob.IsFull() {
+				helpers.LockRecord(lob.ID, lob)
 				lob.State = models.LobbyStateReadyingUp
 				lob.Save()
+				helpers.UnlockRecord(lob.ID, lob)
+
+				go func() {
```

Comment:
I'd like this to be in a separate function because it's not really LobbyJoinHandler's responsibility. Then you can just do `go ReadyUpTimeoutCheck()`. Also, you should synchronize lobby changes in here as well.